### PR TITLE
PLANNER-2145: Remove workaround in flight example solver manager

### PR DESCRIPTION
**Pull request by Christopher-Chianelli**
https://issues.redhat.com/browse/PLANNER-2145

Also fixed errors that were introduced from changes to the tasks
api.

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

**WARNING! Please make sure you are opening your PR against `master` branch!**

- [x] You have read the [contributors guide](https://github.com/kiegroup/kogito-runtimes#contributing-to-kogito)
- [x] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [x] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

**Comment by triceo** [comment]
I have removed myself from the reviewers as I am not familiar with this code.

**Comment by Christopher-Chianelli** [comment]
@triceo Fair enough; it just a long due refactor to use the "new" SolverManager method that handles both best solutions and final best solutions events. Due to changes in Kogito API, I also needed to make some bugs fixes from the API changes in the frontend.

File: process-optaplanner-quarkus/src/main/java/org/kie/kogito/examples/service/FlightSeatingSolveService.java [hunk 1]
```diff
@@ -39,23 +36,17 @@
...
+                                         process.instances().findById(id).ifPresent(pi -> {
+                                             pi.send(Sig.of("newSolution", bestSolution));
+                                         });
+                                     }, finalBestSolution -> {
+                                         process.instances().findById(id).ifPresent(pi -> {
+                                             pi.send(Sig.of("solvingTerminated", finalBestSolution));
+                                         });
+                                     },
+                                     (message, exception) -> {
+                                         process.instances().findById(id).ifPresent(ProcessInstance::abort);
```
Comment (ge0ffrey) [inline process-optaplanner-quarkus/src/main/java/org/kie/kogito/examples/service/FlightSeatingSolveService.java:49]: If an error occurs, does the user somehow get told in the UI that it did? (and kind of of error it was?)
Or does the error get swept under the carpet?

**Review by ge0ffrey — APPROVED** [review]

File: process-optaplanner-quarkus/src/main/java/org/kie/kogito/examples/service/FlightSeatingSolveService.java (hunk 1 above)
Comment (Christopher-Chianelli) [reply process-optaplanner-quarkus/src/main/java/org/kie/kogito/examples/service/FlightSeatingSolveService.java:49]: In this case, we would need to save the exception somewhere as this exception will happen AFTER the REST call returns (so even if we did throw it, the REST call have no way of knowing). The flight will disappear from the UI. So we would need to poll a new REST endpoint for errors.

Comment (Christopher-Chianelli) [reply process-optaplanner-quarkus/src/main/java/org/kie/kogito/examples/service/FlightSeatingSolveService.java:49]: (for errors thrown during REST call invocation, those are shown in the UI already; for an error that occurs in this place, no exception is shown in the UI as it isn't thrown + it requires a new REST endpoint to work (or does Kogito generate an process aborted REST endpoint?)

**Comment by Christopher-Chianelli** [comment]
jenkins retest this

**Pull request merged**